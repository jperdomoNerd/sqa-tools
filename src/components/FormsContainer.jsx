import { useState } from 'react';
import BlueForm from './BlueForm';

export function defineFieldsToDisplayBlueForm (_applicationValue) {
  if (
    ['POS Legacy', 'Cashiering Legacy',
      'POS Reports Legacy', 'Bridge API'].find(
        application => application === _applicationValue
  )) {
    return {
      method: { isShow: true },
      transaction: { isShow: true },
      responseType: { isShow: false },
      tokenType: { isShow: false },
      siteVerify: { isShow: false }
    };
  }
  if (_applicationValue === 'Webpay') {
    return {
      method: { isShow: false },
      transaction: { isShow: true },
      responseType: { isShow: true },
      tokenType: { isShow: false },
      siteVerify: { isShow: false }
    };
  }
  if (_applicationValue === 'SWP Cards'
    || _applicationValue === 'SWP Checks') {
      return {
        method: { isShow: false },
        transaction: { isShow: false },
        responseType: { isShow: false },
        tokenType: { isShow: true },
        siteVerify: { isShow: true }
      }
  }
  if (_applicationValue === 'ViewProcess') {
    return {
      method: { isShow: false },
      transaction: { isShow: false },
      responseType: { isShow: false },
      tokenType: { isShow: false },
      siteVerify: { isShow: false }
    }
  }
  return {
    method: { isShow: false },
    transaction: { isShow: false },
    responseType: { isShow: false },
    tokenType: { isShow: false },
    siteVerify: { isShow: false }
  }
}
const urlSegmented = {
  protocol: 'https://',
  subdomain: '',
  domain: 'cenpos.',
  extension: 'net/',
  restOfFolders: ''
}
function defineSubdomain(environment) {
  if (environment === 'https://webtest.cenpos.net/webpay/v7/html5/') return 'webtest.';
  if (environment === 'https://webqa.cenpos.net/webpay/v7/html5/') return 'webqa.';
  if (environment === 'https://webstaging.cenpos.net/webpay/v7/html5/') return 'webstaging.';
  if (environment === 'https://webstaging2.cenpos.net/webpay/v7/html5/') return 'webstaging2.';
  if (environment === 'https://www4.cenpos.net/webpay/v7/html5/') return 'wwww4.';
  if (environment === 'https://elavonwww.cenpos.net/webpay/v7/html5/') return 'elavonwww.';
  if (environment === 'https://elavon.cenpos.net/webpay/v7/html5/') return 'elavon.';
  return '';
}
function defineRestOfFolders(application) {
  if (application === 'POS Legacy') return 'posintegration/posintegration-html5/';
  if (application === 'Cashiering Legacy') return 'cashiering-html5/';
  if (application === 'POS Reports Legacy') return 'posintegration/posintegration-html5-reports/';
  if (application === 'SWP Cards') return 'simplewebpay/cards/';
  if (application === 'SWP Checks') return 'simplewebpay/checks/';
  if (application === 'Bridge API') return 'spa/app/bridge-api/';
  if (application === 'Webpay') return 'webpay/v7/html5/';
  if (application === 'ViewProcess') return 'viewprocess/';
  return '';
}
export function defineEndpointUrl(enviroment, application) {
  urlSegmented.subdomain = defineSubdomain(enviroment);
  urlSegmented.restOfFolders = defineRestOfFolders(application);
  return `${urlSegmented.protocol}${urlSegmented.subdomain}${urlSegmented.domain}${urlSegmented.extension}${urlSegmented.restOfFolders}`;
}

const blueFormData = {
  application: {
    value: 'POS Legacy',
    isShow: true
  },
  environment: {
    value: 'https://webtest.cenpos.net/webpay/v7/html5/',
    isShow: true
  },
  endpoint: {
    value: '',
    isShow: true
  },
  tokenType: {
    value: 'none',
    isShow: false
  },
  siteVerify: {
    value: 'genericcontroller',
    isShow: false
  },
  method: {
    value: 'POST',
    isShow: true
  },
  transaction: {
    value: 'sale',
    isShow: true
  },
  responseType: {
    value: 'none',
    isShow: false
  }
};

const FormsContainer = ({ handleClick }) => {
  // State
  const [blueForm, setBlueForm] = useState(blueFormData);

  // Events
  const applicationHandleChange = event => {
    const fieldsToShow = defineFieldsToDisplayBlueForm(event.target.value);
    let copyBlueForm = blueForm;
    copyBlueForm.application.value = event.target.value;
    copyBlueForm.method.isShow = fieldsToShow.method.isShow;
    copyBlueForm.transaction.isShow = fieldsToShow.transaction.isShow;
    copyBlueForm.responseType.isShow = fieldsToShow.responseType.isShow;
    copyBlueForm.tokenType.isShow = fieldsToShow.tokenType.isShow;
    copyBlueForm.siteVerify.isShow = fieldsToShow.siteVerify.isShow;
    setBlueForm({ ...copyBlueForm });
  }

  const environmentHandleChange = event => {
    setBlueForm({
      ...blueForm,
      environment: { ...blueForm.environment, value: event.target.value }
    });
  }

  const endpointHandleChange = event => {
    setBlueForm({
      ...blueForm,
      endpoint: { ...blueForm.endpoint, value: event.target.value }
    });
  }

  const methodHandleChange = event => {
    setBlueForm({
      ...blueForm,
      method: { ...blueForm.method, value: event.target.value }
    });
  }

  const transactionHandleChange = event => {
    setBlueForm({
      ...blueForm,
      transaction: { ...blueForm.transaction, value: event.target.value }
    });
  }

  const responseTypeHandleChange = event => {
    setBlueForm({
      ...blueForm,
      responseType: { ...blueForm.responseType, value: event.target.value }
    });
  }

  const tokenTypeHandleChange = event => {
    setBlueForm({
      ...blueForm,
      tokenType: { ...blueForm.tokenType, value: event.target.value }
    });
  }

  const siteVerifyHandleChange = event => {
    setBlueForm({
      ...blueForm,
      siteVerify: { ...blueForm.siteVerify, value: event.target.value }
    });
  }

  // Functions
  const setUrl = () => {
    let endpointUrl = defineEndpointUrl(
      blueForm.environment.value, blueForm.application.value);
    if (blueForm.environment.value === 'custom') {
      endpointUrl = blueForm.endpoint.value;
    }
    handleClick(endpointUrl);
  }

  return (
    <section className="forms-container">
      <button className="forms-container__button" onClick={setUrl}>Load Object                     </button>
      <BlueForm
        blueForm={blueForm}
        applicationHandleChange={applicationHandleChange}
        environmentHandleChange={environmentHandleChange}
        endpointHandleChange={endpointHandleChange}
        methodHandleChange={methodHandleChange}
        transactionHandleChange={transactionHandleChange}
        responseTypeHandleChange={responseTypeHandleChange}
        tokenTypeHandleChange={tokenTypeHandleChange}
        siteVerifyHandleChange={siteVerifyHandleChange}
      />
      {/* <GreenForm />
      <RedForm /> */}
    </section>
  );
}

export default FormsContainer;
