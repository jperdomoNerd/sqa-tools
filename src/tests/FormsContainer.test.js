import renderer from 'react-test-renderer';
import FormsContainer, { defineFieldsToDisplayBlueForm, defineEndpointUrl } from '../components/FormsContainer';

describe('FormsContainer Component', () => {
  
  it('<FormsContainer /> renders correctly', () => {
    const tree = renderer.create(<FormsContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('Functions of the FormsContainer component', () => {
  
  describe('defineFieldsToDisplayBlueForm', () => {

    test('Does not show any fields (returns all properties false)', () => {
      const result = defineFieldsToDisplayBlueForm('ViewProcess');
      expect(result).toEqual({
        method: { isShow: false },
        transaction: { isShow: false },
        responseType: { isShow: false },
        tokenType: { isShow: false },
        siteVerify: { isShow: false }
      });
    });

    test('Only show method and transaction', () => {
      const result = defineFieldsToDisplayBlueForm('POS Reports Legacy');
      expect(result).toEqual({
        method: { isShow: true },
        transaction: { isShow: true },
        responseType: { isShow: false },
        tokenType: { isShow: false },
        siteVerify: { isShow: false }
      });
    });

    test('Only show tokenType and siteVerify', () => {
      const result = defineFieldsToDisplayBlueForm('SWP Checks');
      expect(result).toEqual({
        method: { isShow: false },
        transaction: { isShow: false },
        responseType: { isShow: false },
        tokenType: { isShow: true },
        siteVerify: { isShow: true }
      });
    });

  });

  describe('defineEndpointUrl', () => {
    
    test('Return https://webtest.cenpos.net/simplewebpay/cards/', () => {
      const result = defineEndpointUrl('https://webtest.cenpos.net/webpay/v7/html5/', 'SWP Cards');
      expect(result).toEqual('https://webtest.cenpos.net/simplewebpay/cards/');
    });
    
    test('Return https://webqa.cenpos.net/simplewebpay/checks/', () => {
      const result = defineEndpointUrl('https://webqa.cenpos.net/webpay/v7/html5/', 'SWP Checks');
      expect(result).toEqual('https://webqa.cenpos.net/simplewebpay/checks/');
    });

    test('Return https://webstaging.cenpos.net/simplewebpay/checks/', () => {
      const result = defineEndpointUrl('https://webstaging.cenpos.net/webpay/v7/html5/', 'SWP Checks');
      expect(result).toEqual('https://webstaging.cenpos.net/simplewebpay/checks/');
    });

    test('Return https://webtest.cenpos.net/cashiering-html5/', () => {
      const result = defineEndpointUrl('https://webtest.cenpos.net/webpay/v7/html5/', 'Cashiering Legacy');
      expect(result).toEqual('https://webtest.cenpos.net/cashiering-html5/');
    });

  });

});
