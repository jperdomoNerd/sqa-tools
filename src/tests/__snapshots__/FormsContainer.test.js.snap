// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormsContainer Component <FormsContainer /> renders correctly 1`] = `
<section
  className="forms-container"
>
  <button
    className="forms-container__button"
    onClick={[Function]}
  >
    Load Object                     
  </button>
  <form
    className="form form--blue"
  >
    <div
      className="form__input-container"
    >
      <label
        className="form__label"
      >
        Application:
      </label>
      <select
        className="form__input"
        onChange={[Function]}
        value="POS Legacy"
      >
        <option
          value="POS Legacy"
        >
          POS Legacy
        </option>
        <option
          value="Cashiering Legacy"
        >
          Cashiering Legacy
        </option>
        <option
          value="POS Reports Legacy"
        >
          POS Reports Legacy
        </option>
        <option
          value="Bridge API"
        >
          Bridge API
        </option>
        <option
          value="Webpay"
        >
          Webpay
        </option>
        <option
          value="SWP Cards"
        >
          SWP Cards
        </option>
        <option
          value="SWP Checks"
        >
          SWP Checks
        </option>
        <option
          value="ViewProcess"
        >
          ViewProcess
        </option>
      </select>
    </div>
    <div
      className="form__input-container"
    >
      <label
        className="form__label"
      >
        Environment:
      </label>
      <select
        className="form__input"
        onChange={[Function]}
        value="https://webtest.cenpos.net/webpay/v7/html5/"
      >
        <option
          value="https://webtest.cenpos.net/webpay/v7/html5/"
        >
          Dev
        </option>
        <option
          value="https://webqa.cenpos.net/webpay/v7/html5/"
        >
          QA
        </option>
        <option
          value="https://webstaging.cenpos.net/webpay/v7/html5/"
        >
          Staging: KNX
        </option>
        <option
          value="https://webstaging2.cenpos.net/webpay/v7/html5/"
        >
          Staging: OBC
        </option>
        <option
          value="https://www4.cenpos.net/webpay/v7/html5/"
        >
          Prod: LB
        </option>
        <option
          value="https://elavonwww.cenpos.net/webpay/v7/html5/"
        >
          Prod: KNX
        </option>
        <option
          value="https://elavon.cenpos.net/webpay/v7/html5/"
        >
          Prod: OBC
        </option>
        <option
          value="custom"
        >
          Custom
        </option>
      </select>
    </div>
    <div
      className="form__input-container"
    >
      <label
        className="form__label"
      >
        Method:
      </label>
      <select
        className="form__input"
        onChange={[Function]}
        value="POST"
      >
        <option
          value="POST"
        >
          POST
        </option>
        <option
          value="GET"
        >
          GET
        </option>
      </select>
    </div>
    <div
      className="form__input-container"
    >
      <label
        className="form__label"
      >
        Transaction:
      </label>
      <select
        className="form__input"
        onChange={[Function]}
        value="sale"
      >
        <option
          value="sale"
        >
          sale
        </option>
        <option
          value="achcredit"
        >
          achcredit
        </option>
        <option
          value="achdebit"
        >
          achdebit
        </option>
        <option
          value="activate"
        >
          activate
        </option>
        <option
          value="auth"
        >
          auth
        </option>
        <option
          value="createtoken"
        >
          createtoken
        </option>
        <option
          value="credit"
        >
          credit
        </option>
        <option
          value="ebppinvoice"
        >
          ebppinvoice
        </option>
        <option
          value="force"
        >
          force
        </option>
        <option
          value="guarantee"
        >
          guarantee
        </option>
        <option
          value="inquire"
        >
          inquire
        </option>
        <option
          value="moto"
        >
          moto
        </option>
        <option
          value="reauth"
        >
          reauth
        </option>
        <option
          value="recurringauth"
        >
          recurringauth
        </option>
        <option
          value="recurringcredit"
        >
          recurringcredit
        </option>
        <option
          value="recurringsale"
        >
          recurringsale
        </option>
        <option
          value="reload"
        >
          reload
        </option>
        <option
          value="repeatauth"
        >
          repeatauth
        </option>
        <option
          value="repeatsale"
        >
          repeatsale
        </option>
        <option
          value="return"
        >
          return
        </option>
        <option
          value="reversal"
        >
          reversal
        </option>
        <option
          value="salemoto"
        >
          salemoto
        </option>
        <option
          value="signature"
        >
          signature
        </option>
        <option
          value="tokencheck"
        >
          tokencheck
        </option>
        <option
          value="usechecktoken"
        >
          usechecktoken
        </option>
        <option
          value="voiceauth"
        >
          voiceauth
        </option>
        <option
          value="void"
        >
          void
        </option>
      </select>
    </div>
  </form>
</section>
`;
